### ISSUE TITLE
JDBC Driver add support for OpenJDK 17

### ISSUE THREAD

**Issue by wise-coders** [apache/ignite#10449]
I have tested the JDBC driver under OpenJDK17, and does not work. I tested ignite-core-2.14.0.jar.
Same driver works fine with OpenJDK15.

**Comment by ptupitsyn** [comment]
1. Which JDBC driver did you try - "thin" or full?
2. What is the error?

**Comment by wise-coders** [comment]
I tried the full. They are part of the same library, ignite-core-2.14.0, right?
I am getting an empty ( null ) connection. Changing to OpenJDK15 works fine.

**Comment by ptupitsyn** [comment]
Did you provide all required JVM flags?

```
--add-opens=jdk.management/com.sun.management.internal=ALL-UNNAMED \
--add-opens=java.base/jdk.internal.misc=ALL-UNNAMED \
--add-opens=java.base/sun.nio.ch=ALL-UNNAMED \
--add-opens=java.management/com.sun.jmx.mbeanserver=ALL-UNNAMED \
--add-opens=jdk.internal.jvmstat/sun.jvmstat.monitor=ALL-UNNAMED \
--add-opens=java.base/sun.reflect.generics.reflectiveObjects=ALL-UNNAMED \
--add-opens=java.base/java.io=ALL-UNNAMED \
--add-opens=java.base/java.nio=ALL-UNNAMED \
--add-opens=java.base/java.util=ALL-UNNAMED \
--add-opens=java.base/java.lang=ALL-UNNAMED
```

https://ignite.apache.org/docs/latest/quick-start/java#running-ignite-with-java-11-or-later

**Comment by wise-coders** [comment]
No, I don't set any flags. I use the library as JDBC driver, to connect to the database. The library is loaded in a database client, Dbschema.

**Comment by ivandasch** [comment]
@wise-coders Hi!
It is advisable to use a jdbc-thin one. Could you please share here your connection string?

**Comment by wise-coders** [comment]
Where can I find the JDBC thin driver? I cannot find how to download it. I searched this page first, and Google:
https://ignite.apache.org/docs/latest/SQL/JDBC/jdbc-driver

I used apache-core-2.14.0.jar as the single one I found in Maven Central.
The apache-core-3.0.0 seems to have no Driver class inside. Is this right?

The URL: jdbc:ignite:thin://{HOST}[:{PORT}]

**Comment by wise-coders** [comment]
Here is the URL:
jdbc:ignite:thin://{HOST}[:{PORT}]
port defaults to 10800
